internal/cli/capact/components.go

```diff
@@ -320,7 +321,31 @@ func (n *Neo4j) InstallUpgrade(ctx context.Context, version string) (*release.Re
 func (a *Argo) InstallUpgrade(ctx context.Context, version string) (*release.Release, error) {
 	upgradeCli := a.upgradeAction(version)
 
-	values := tools.MergeMaps(map[string]interface{}{}, a.Overrides)
+	values := make(map[string]interface{})
+	minioValues := make(map[string]interface{})
+	values["minio"] = minioValues
+
+	accessKeyMap := make(map[string]interface{})
+	minioValues["accessKey"] = accessKeyMap
+
+	secretKeyMap := make(map[string]interface{})
+	minioValues["secretKey"] = secretKeyMap
+
+	k8sClient, _ := a.configuration.KubernetesClientSet()
+	s, err := k8sClient.CoreV1().Secrets(a.opts.Namespace).Get(ctx, a.ReleaseName+"-minio", metav1.GetOptions{})
+
+	if apierrors.IsNotFound(err) {
+		// Chart does not exist, so we have to generate and set the credentials
```

Comment:
This assumes the secret is always created along Minio installation. What if someone uses `existingSecret` with accessKey and secretKey? Maybe we should use it this way on our CI?

Reply:
Okay, I checked the current behavior and we need to just pass access key and secret key during the installation if user hasn't provided existing secret.

The commands:
```
helm install my-release bitnami/minio --set accessKey.password=423432432423 --set secretKey.password=231213123123
helm upgrade my-release bitnami/minio
```
work flawlessly.


So, please refactor it to something like:

```go
func (a *Argo) InstallUpgrade(ctx context.Context, version string) (*release.Release, error) {
	upgradeCli := a.upgradeAction(version)


	values := a.Overrides
	// TODO: Split it to separate func
	if (existingSecret is not provided and this is very first Argo installation) { // TODO: Change to real condition instead of pseudocode 

		// Chart does not exist, so we have to generate and set the credentials

		var accessKey, secretKey string
		var err error

		
		accessKey, err = util.CryptoRandomAlphaNumeric(10)
		// TODO: handle error
		secretKey, err = util.CryptoRandomAlphaNumeric(40)
		// TODO: handle error


		values := map[string]interface{}{
			"minio": map[string]interface{}{
				"accessKey": map[string]interface{}{
					"password": accessKey,
				},
				"secretKey": map[string]interface{}{
					"password": secretKey,
				},
			},
		}


		values = tools.MergeMaps(values, a.Overrides)
	}
	
	return a.runUpgrade(upgradeCli, values)
}
```